sympy/matrices/tests/test_matrices.py

```diff
@@ -259,6 +260,19 @@ def test_power():
     raises(ValueError, lambda: An.subs(n, -2).doit())
     assert An * An == A**(2*n)
 
+    # concretizing behavior for non-integer and complex powers
+    A = Matrix ([[0,0,0],[0,0,0],[0,0,0]])
+    n = Symbol('n', integer=True, positive=True)
+    assert A**n == A
+    n = Symbol('n', integer=True, nonnegative=True)
+    assert A**n != A
```

Comment:
This test should be more explicit. There are many possible incorrect evaluations of `A**n` that would pass here. We should test for exactly what *is* supposed to be returned.

Also see comments here: https://github.com/sympy/sympy/pull/17206/files#r304160029

Reply:
`A = Matrix ([[0,0,0],[0,0,0],[0,0,0]])`, the test is that a zero matrix to any positive power is zero but to a nonnegative should not return zero because it COULD be identity if exponent is substituted as 0 in the future.

Reply:
Yes but we need to test what it should be. And we should probably also test that substituting n=0 gives the identity.